### PR TITLE
added run conditional for GitHub workflows

### LINKED ISSUE #306: only execute runners when you've edited relevant code

### PR DESCRIPTION
**Pull request by owenrhysthomas**
This PR added path conditionals when a workflow is triggered by a pull request, meaning that the associated workflows only run when the associated with a certain path. An extension of this could be to split some of the workflow files up further, for example give ph-debug its own runner so the ph-debug runners only execute when ph-debug files are changed, not when any file in the adapter/ph or adapter/cd or adapter/ph-debug folder are edited.

### CODE REVIEW

**Comment by owenrhysthomas** [comment]
I'm not 100% sure what all the various dependencies between various modules are. I did my best to make sure that associated runners will execute at the appropriate time, but I'm not positive I did that, perhaps people working on the different pieces before we commit this just to make sure its what they want/need/expect.

File: .github/workflows/go-visaservice.yml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
   push:
     branches: [ "main" ]
   pull_request:
+    paths:
+      - visaservice/mods/**
+      - visaservice/core/**
```
Comment (mkolehmainen) [inline .github/workflows/go-visaservice.yml:12]: add `visaservice/thrift/**`

File: .github/workflows/rust-node.yml [hunk 2]
```diff
@@ -4,6 +4,8 @@ on:
   push:
     branches: [ "main" ]
   pull_request:
+    paths: 
+      - node/**
```
Comment (mkolehmainen) [inline .github/workflows/rust-node.yml:8]: add `visaservice/thrift/**`

File: .github/workflows/rust-visaservice.yml [hunk 3]
```diff
@@ -4,7 +4,9 @@ on:
   push:
     branches: [ "main" ]
   pull_request:
-
+    paths:
+      - visaservice/cli/**
```
Comment (mkolehmainen) [inline .github/workflows/rust-visaservice.yml:8]: add `visaservice/thrift/**`

File: .github/workflows/rust-zprext.yml [hunk 4]
```diff
@@ -1,9 +1,11 @@
-name: Rust-cbpfrs
+name: Rust-zprext
 
 on:
   push:
     branches: [ "main" ]
   pull_request:
+    paths:
```
Comment (mkolehmainen) [inline .github/workflows/rust-zprext.yml:7]: If you alter zpr-ext, you better also recompile:
- adapter/cd
- adapter/ph
- adapter/ph-debug
- node

File: .github/workflows/rust-adapter.yml [hunk 5]
```diff
@@ -4,6 +4,10 @@ on:
   push:
     branches: [ "main" ]
   pull_request:
+    paths:
+      - adapter/ph/**
```
Comment (mkolehmainen) [inline .github/workflows/rust-adapter.yml:8]: note that if you alter ph, you should recompile node.

**Review by nonrenewable — APPROVED** [review]
I really like the concept!! Thank you for doing this 🙇 

But i don't know all the dependencies so either @svjaffer and @cpacejo also need to look/guarantee, or just merge it in and as things get weird - we know about this and fix it as we go!

File: .github/workflows/rust-adapter.yml [hunk 6]
```diff
@@ -4,6 +4,11 @@ on:
...
+      - adapter/cd/**
+      - adapter/ph-debug/**
+      - zpr-ext/**
```
Comment (cpacejo) [inline .github/workflows/rust-adapter.yml:11]: `cbpf-rs` and `cslab` also are dependencies of both the adapter and node

**Review by cpacejo — APPROVED** [review]
Awesome!!  Just a couple extra dependencies to add for the adapter and node

**Pull request merged**